### ISSUE TITLE
Sonoff Basic Z gone with the last upgrade

### ISSUE THREAD

**Issue by royby12** [ebaauw/homebridge-hue#640]
Hi,

Sonoff BASICZBR3 Is gone from HomeKit with the recent update.
It does appear in Phoscon as a light but not reflected in HomeKit.
These lights are also not initiated when starting homebridge.

That is my conf:

{
		"platform": "Hue",
		"host": "127.0.0.1",
		"users": {
			"00212EFFFF040D49": "F52A32751E"
			},
		"sensors": true,
		"excludeSensorTypes": ["CLIP","Geofence"],
        "lights": true,
        "hueMotionTemperatureHistory": true,
        "forceCt": false
     },

#### Log Messages

```
```

**Comment by ebaauw** [comment]
Please attach the debug dump file.

**Comment by suhajda3** [comment]
Same for me. Tasmota devices are no longer working after updating homebridge-hue to v0.11.56.
I tried to update my devices firmware to the latest Tasmota 8.1.0 but still doesn't work.

**Comment by royby12** [comment]
Log attached

[homebridge.log](https://github.com/ebaauw/homebridge-hue/files/4270634/homebridge.log)

**Comment by ebaauw** [comment]
Ah, they report `type` as `”Unknown”`.  I just blacklisted these because of #635.

**Comment by royby12** [comment]
Hummm... Phoscon identify them as BASICZBR3:
![E6BA2F83-E971-4BA2-917D-432960A08671](https://user-images.githubusercontent.com/9537054/75612385-8a494380-5b2b-11ea-85ff-670c70d37f62.jpeg)

**Comment by ebaauw** [comment]
Could you try beta v0.11.57-0?  Install thru `sudo npm -g homebridge-hue@beta`.

**Comment by royby12** [comment]
👍🏽 Confirmed.
It’s working again.

**Comment by suhajda3** [comment]
For me it still doesn't work. :(

I edited the HueBridge.js manually based on the commit.

[debug.log](https://github.com/ebaauw/homebridge-hue/files/4270677/debug.log)

**Comment by ebaauw** [comment]
@suhajda3, I don't think your issue is related.  You seem to be exposing one real Hue bridge and four virtual bridges.  The real bridge named "Philips hue" and the virtual bridge named "hue emulator" seem to be properly exposed, with "Christmas tree" and "Countertop".

The three virtual bridges also named "Hue bridge" seem to be invalid, in that they're not exposing the proper resources.  In particular, they don't expose `/groups`, causing homebridge-hue to skip discovery of their device(s).
```
Feb 29 18:59:59 raspberrypi homebridge[18335]: [2/29/2020, 18:59:59] [Hue] Philips hue: Cannot set property '0' of undefined
```
Also, one (or more) of the virtual bridges exposes an invalid (or missing?) value for `/config/proxyaddress`.  Because of this, no dump file is created.
```
Feb 29 19:00:00 raspberrypi homebridge[18335]: [2/29/2020, 19:00:00] [Hue] TypeError: Cannot read property 'split' of undefined
Feb 29 19:00:00 raspberrypi homebridge[18335]:     at HuePlatform.maskHost (/usr/lib/node_modules/homebridge-hue/lib/HuePlatform.js:402:20)
Feb 29 19:00:00 raspberrypi homebridge[18335]:     at HuePlatform.dump (/usr/lib/node_modules/homebridge-hue/lib/HuePlatform.js:382:42)
Feb 29 19:00:00 raspberrypi homebridge[18335]:     at HuePlatform.accessories (/usr/lib/node_modules/homebridge-hue/lib/HuePlatform.js:317:16)
Feb 29 19:00:00 raspberrypi homebridge[18335]:     at processTicksAndRejections (internal/process/task_queues.js:94:5)
```

> I tried to update my devices firmware to the latest Tasmota 8.1.0 but still doesn't work.

Which devices did you update?  Did the update break compatibility?  Could you try a `ph -H 192.x.x.x get /` on one of the updated devices and on the "hue emulator"?

**Comment by suhajda3** [comment]
The three virtual bridges (that seem to be invalid) are the updated Tasmota devices. I just updated them from 6.7.1 to 8.1.0. (I have followed the correct migration path)

They worked perfectly before with Sonoff-Tasmota 6.7.1 and the previous version of homebridge-hue.

Here are the results from ph:
```
root@raspberrypi:/home/pi# ph -H 192.168.99.13 get /
ph: fatal: missing username - press link button and run "ph -H 192.168.99.13 createuser"
root@raspberrypi:/home/pi# ph -H 192.168.99.13 createuser
"df662a"
root@raspberrypi:/home/pi# ph -H 192.168.99.13 get /
[
  {
    "success": {
      "username": "df662a"
    }
  }
]
```

**Comment by suhajda3** [comment]
This is the config:
```
# ph -H 192.168.99.13 config
{
  "name": "Philips hue",
  "mac": "DC:4F:22:DF:66:2A",
  "dhcp": true,
  "ipaddress": "192.168.99.13",
  "netmask": "255.255.255.0",
  "gateway": "192.168.99.1",
  "proxyaddress": "none",
  "proxyport": 0,
  "bridgeid": "DC4F22FFFEDF662A",
  "UTC": "2020-02-29T20:19:49",
  "whitelist": {
    "df662a": {
      "last use date": "2020-02-29T20:19:49",
      "create date": "2020-02-29T20:19:49",
      "name": "Remote"
    }
  },
  "swversion": "01041302",
  "apiversion": "1.17.0",
  "swupdate": {
    "updatestate": 0,
    "url": "",
    "text": "",
    "notify": false
  },
  "linkbutton": false,
  "portalservices": false
}
```

**Comment by ebaauw** [comment]
> Here are the results from ph:

I don’t think the Tasmota supports usernames.  Just use `ph -u whatever` to specify a (fake) username on the command line.  It looks like the GET / is returning the status of the creation of the username.  That would be problematic.

> They worked perfectly before with Sonoff-Tasmota 6.7.1

Can you revert to that version?

**Comment by suhajda3** [comment]
Downgrade is not officially supported. I just bricked one device... :(

But I could downgrade another one with a different technique.

Unfortunately the result is the same:
```
root@raspberrypi:/home/pi# ph -H 192.168.99.12 get /
[
  {
    "success": {
      "username": "cc6368"
    }
  }
]
root@raspberrypi:/home/pi# ph -H 192.168.99.12 -u cc6368 get /
[
  {
    "success": {
      "username": "cc6368"
    }
  }
]
```

**Comment by suhajda3** [comment]
Alexa can still control the remaining two devices (one with downgraded version 6.7.1 and one with the updated 8.1.0).

**Comment by ebaauw** [comment]
I'm afraid I cannot help you.  I don't know Tasmota nor Alexa.

**Comment by suhajda3** [comment]
Alexa uses the Hue emulation of Tasmota so Hue emulation should be working "correctly" on the device.

Bad day for updating. Should have left as it was...

**Comment by suhajda3** [comment]
If I directly specify a path I can get information from the device:
```
# ph -H 192.168.99.10 -u api get /lights
{
  "200980881": {
    "state": {
      "on": false,
      "bri": 254,
      "alert": "none",
      "effect": "none",
      "reachable": true
    },
    "type": "Extended color light",
    "name": "Exhaust fan",
    "modelid": "LCT007",
    "uniqueid": "60:01:94:bf:ab:99:00:11-1",
    "swversion": "5.50.1.19085"
  }
}
```

I get the same output for both firmware versions.

**Comment by suhajda3** [comment]
I don't know if this is expected but if I specify an invalid  path I get all the paths. :)

```
# ph -H 192.168.99.12 -u api get /abc
{
  "lights": {
    "214316673": {
      "state": {
        "on": false,
        "bri": 254,
        "alert": "none",
        "effect": "none",
        "reachable": true
      },
      "type": "Extended color light",
      "name": "Salt lamp",
      "modelid": "LCT007",
      "uniqueid": "60:01:94:cc:63:68:00:11-1",
      "swversion": "5.50.1.19085"
    }
  },
  "groups": {},
  "schedules": {},
  "config": {
    "name": "Philips hue",
    "mac": "60:01:94:CC:63:68",
    "dhcp": true,
    "ipaddress": "192.168.99.12",
    "netmask": "255.255.255.0",
    "gateway": "192.168.99.1",
    "proxyaddress": "none",
    "proxyport": 0,
    "bridgeid": "600194FFFECC6368",
    "UTC": "2020-02-29T23:02:36",
    "whitelist": {
      "cc6368": {
        "last use date": "2020-02-29T23:02:36",
        "create date": "2020-02-29T23:02:36",
        "name": "Remote"
      }
    },
    "swversion": "01041302",
    "apiversion": "1.17.0",
    "swupdate": {
      "updatestate": 0,
      "url": "",
      "text": "",
      "notify": false
    },
    "linkbutton": false,
    "portalservices": false
  }
}
```

**Comment by ebaauw** [comment]
> I don't know if this is expected but if I specify an invalid path I get all the paths. 

You would have to ask the Tasmota people.

The good news: this seems like a bug more than a breaking change.  To create a user (API key) you actually do a POST to `/`.  I would seem like the Tasmota mistakingly handles GET `/` as POST `/`.

**Comment by suhajda3** [comment]
I hope @s-hadinger from Tasmota can step-in and help resolve this issue.

**Comment by ebaauw** [comment]
Could you check what's the output of `ph -H ... -u x get /groups/0`?  Also the full state?

**Comment by s-hadinger** [comment]
Tasmota does not care about user and doesn't support Hue groups. We have implemented the absolute minimal support for Alexa. 

Tasmota will only respond to group 0 requests and send the list of lights.

**Comment by ebaauw** [comment]
Thanks for chiming in, @s-hadinger.

> Tasmota does not care about user and doesn't support Hue groups. We have implemented the absolute minimal support for Alexa.

That's cool, Tasmota used to support enough of the Hue API to be supported by homebridge-hue,  see #512.  However, with the newer firmware, the request to get the full state (GET `/api/`_username_)  seems to be broken; the response looks like the response to a request to create a user (POST `/api`).

I'm sorry, I cannot provide more info; I don't have any Sonoff devices and any know of Tasmota from the other issue.

**Comment by suhajda3** [comment]
I have run the requested commands.

```
pi@raspberrypi:~ $ curl http://192.168.99.12/api/x
{"lights":{"214316673":{"state":{"on":false,"alert":"none","effect":"none","reachable":true},"type":"Extended color light","name":"Salt lamp","modelid":"LCT007","uniqueid":"60:01:94:cc:63:68:00:11-1","swversion":"5.50.1.19085"}},"groups":{},"schedules":{},"config":{"name":"Philips hue","mac":"60:01:94:CC:63:68","dhcp":true,"ipaddress":"192.168.99.12","netmask":"255.255.255.0","gateway":"192.168.99.1","proxyaddress":"none","proxyport":0,"bridgeid":"600194FFFECC6368","UTC":"2020-03-01T13:11:47","whitelist":{"cc6368":{"last use date":"2020-03-01T13:11:47","create date":"2020-03-01T13:11:47","name":"Remote"}},"swversion":"01041302","apiversion":"1.17.0","swupdate":{"updatestate":0,"url":"","text":"","notify": false},"linkbutton":false,"portalservices":false}}
pi@raspberrypi:~ $ curl http://192.168.99.12/api/x/
[{"success":{"username":"cc6368"}}]
pi@raspberrypi:~ $ curl http://192.168.99.12/api/x/groups
{}
pi@raspberrypi:~ $ curl http://192.168.99.12/api/x/groups/0
{"name":"Group 0","lights":["1"],"type":"LightGroup","action":{"on":false,"alert":"none","effect":"none","reachable":true}}
```

**Comment by ebaauw** [comment]
Thanks.  I can confirm the issue is with homebridge-hue.  I will come up with a fix shortly.

**Comment by ebaauw** [comment]
@suhajda3, could you try beta v0.11.57-1?  Install thru `sudo npm -g homebridge-hue@beta`.

**Comment by suhajda3** [comment]
Works perfectly! 👍 

(Just want to let you know that I reflashed the bricked device and that is back to normal as well)

Thanks again for the 🥇 support!

**Comment by ebaauw** [comment]
Cool.  I'll release a proper version later today.

**Comment by ebaauw** [comment]
Published v0.11.57.
